Add 4.4 beta 4 blog post (#1012)
godotengine/godot-website: collections/_article/dev-snapshot-godot-4-4-beta-4.md

---
title: "Dev snapshot: Godot 4.4 beta 4"
excerpt: The 4.4 beta phase nears its end!
categories: [pre-release]
author: Thaddeus Crews
image: /storage/blog/covers/dev-snapshot-godot-4-4-beta-4.webp
image_caption_title: Dawnfolk
image_caption_description: A game by Darenn Keller
date: 2025-02-17 12:00:00
---

> We knew from the beginning that aiming for a late-February  release was an ambitious  timeframe, but the community has risen to the challenge and kept development right on track! As such, you can expect this to be our final beta release of the 4.4 period, with release candidates set to come out shortly after. In practice, this means that all further changes will be *strictly* regression fixes, so the content available here will be largely reflective of the 4.4 release. As always, users are strongly encouraged to test this snapshot to catch the remaining few release blockers.

Reviewer: ```suggestion
We knew from the beginning that aiming for a late-February release was an ambitious timeframe, but the community has risen to the challenge and kept development right on track! As such, you can expect this to be our final beta release of the 4.4 period, with release candidates set to come out shortly after. In practice, this means that all further changes will be *strictly* regression fixes, so the content available here will be largely reflective of the 4.4 release. As always, users are strongly encouraged to test this snapshot to catch the remaining few release blockers.
```

---
title: "Dev snapshot: Godot 4.4 beta 4"
excerpt: The 4.4 beta phase nears its end!
categories: [pre-release]
author: Thaddeus Crews
image: /storage/blog/covers/dev-snapshot-godot-4-4-beta-4.webp
image_caption_title: Dawnfolk
image_caption_description: A game by Darenn Keller
date: 2025-02-17 12:00:00
---

> We knew from the beginning that aiming for a late-Februrary release was an ambnitious timeframe, but the community has risen to the challenge and kept development right on track! As such, you can expect this to be our final beta release of the 4.4 period, with release candidates set to come out shortly after. In practice, this means that all further changes will be *strictly* regression fixes, so the content available here will be largely reflective of the 4.4 release. As always, users are strongly encouraged to test this snapshot to catch the remaining few release blockers.

Reviewer 2: ```suggestion
We knew from the beginning that aiming for a late-February release was an ambitious timeframe, but the community has risen to the challenge and kept development right on track! As such, you can expect this to be our final beta release of the 4.4 period, with release candidates set to come out shortly after. In practice, this means that all further changes will be *strictly* regression fixes, so the content available here will be largely reflective of the 4.4 release. As always, users are strongly encouraged to test this snapshot to catch the remaining few release blockers.
```

Please, consider [supporting the project financially](#support), if you are able. Godot is maintained by the efforts of volunteers and a small team of paid contributors. Your donations go towards sponsoring their work and ensuring they can dedicate their undivided attention to the needs of the project.

[Jump to the **Downloads** section](#downloads), and give it a spin right now, or continue reading to learn more about improvements in this release. You can also [try the **Web editor**](https://editor.godotengine.org/releases/4.4.beta4/) or the **Android editor** for this release. If you are interested in the latter, please request to join [our testing group](https://groups.google.com/g/godot-testers) to get access to pre-release builds.

---

> *The cover illustration is from* [**Dawnfolk**](https://store.steampowered.com/app/2308630/Dawnfolk/?curator_clanid=41324400), *a charmingly dark and minimalist survival city-builder, developed by Darenn Keller and published by Astra Logical! You can buy the game [on Steam](https://store.steampowered.com/app/2308630/Dawnfolk/?curator_clanid=41324400), and follow the developer on [BlueSky](https://bsky.app/profile/darenn.bsky.social) and his [website](https://darenn.github.io/linktree/).*

Reviewer 2: ```suggestion
*The cover illustration is from* [**Dawnfolk**](https://store.steampowered.com/app/2308630/Dawnfolk/?curator_clanid=41324400), *a charmingly dark and minimalist survival city-builder, developed by Darenn Keller and published by Astra Logical! You can buy the game [on Steam](https://store.steampowered.com/app/2308630/Dawnfolk/?curator_clanid=41324400), and follow the developer on [Bluesky](https://bsky.app/profile/darenn.bsky.social) and his [website](https://darenn.github.io/linktree/).*
```

## Highlights

For an overview of what's new overall in Godot 4.4, have a look at the highlights for [4.4 beta 1](/article/dev-snapshot-godot-4-4-beta-1/), which cover a lot of the changes. This blog post only covers the changes between beta 3 and beta 4. This section covers the most relevant changes made since the [beta 3 snapshot](/article/dev-snapshot-godot-4-4-beta-3/), which are largely regression fixes.

> ### And more!

Reviewer 2: Shouldn't be "And more!" unless you've highlighted some others first!

- 2D: Fix `Camera2D` limits drawing ([GH-102868](https://github.com/godotengine/godot/pull/102868)).
- 3D: Add changeable freelook speed in Game Window ([GH-102704](https://github.com/godotengine/godot/pull/102704)).
- 3D: Fix collision reposition with `CSGShape3D` ([GH-102286](https://github.com/godotengine/godot/pull/102286)).
- 3D: Fix stale VoxelGI warnings ([GH-102843](https://github.com/godotengine/godot/pull/102843)).

> - Audio: [AudioStreamGenerator] Add mixing rate presets, update docs ([GH-102691](https://github.com/godotengine/godot/pull/102691)).

Reviewer 2: ```suggestion
- Audio: AudioStreamGenerator: Add mixing rate presets, update docs ([GH-102691](https://github.com/godotengine/godot/pull/102691)).
```

- Audio: Fix Theora video issues ([GH-101958](https://github.com/godotengine/godot/pull/101958)).
- Buildsystem: Separate Android editor artifacts ([GH-102543](https://github.com/godotengine/godot/pull/102543)).

> - C#: Disallow `[ExportToolButton]` on members thay may store the Callable ([GH-102836](https://github.com/godotengine/godot/pull/102836)).

Reviewer 2: ```suggestion
- C#: Disallow `ExportToolButton` on members that may store the Callable ([GH-102836](https://github.com/godotengine/godot/pull/102836)).
```

> - C#: Fix Android mono export with 2 or more cpu architectures fails ([GH-98066](https://github.com/godotengine/godot/pull/98066)).

Reviewer 2: ```suggestion
- C#: Fix Android mono export with 2 or more CPU architectures fails ([GH-98066](https://github.com/godotengine/godot/pull/98066)).
```

- C#: Validate project TFM for Android template exports ([GH-102627](https://github.com/godotengine/godot/pull/102627)).
- Documentation: Add parentheses to method links in online class reference ([GH-102567](https://github.com/godotengine/godot/pull/102567)).
- Editor: Fix accessing UID before first scan ([GH-102513](https://github.com/godotengine/godot/pull/102513)).
- Editor: Fix lag when resizing Floating Game Window ([GH-102618](https://github.com/godotengine/godot/pull/102618)).
- Editor: Return fast for built-in class icon ([GH-101435](https://github.com/godotengine/godot/pull/101435)).
- Editor: Show enum property invalid value in inspector ([GH-102743](https://github.com/godotengine/godot/pull/102743)).
- Editor: Tweak Quick Open theming ([GH-101598](https://github.com/godotengine/godot/pull/101598)).
- GDScript: Fix Plugin Scripts load twice on startup ([GH-102535](https://github.com/godotengine/godot/pull/102535)).

> - GUI: [Android] Add Snackbar UI component ([GH-102590](https://github.com/godotengine/godot/pull/102590)).

Reviewer 2: ```suggestion
- GUI: Android: Add Snackbar UI component ([GH-102590](https://github.com/godotengine/godot/pull/102590)).
```

- GUI: Fix Tree Mouse hover position ([GH-102842](https://github.com/godotengine/godot/pull/102842)).
- GUI: Optimize `Font` calculations by avoiding unnecessary copy-on-write ([GH-102132](https://github.com/godotengine/godot/pull/102132)).

> - Input: [Web] Refactor `mouse_mode` setters in display server ([GH-102719](https://github.com/godotengine/godot/pull/102719)).

Reviewer 2: ```suggestion
- Input: Web: Refactor `mouse_mode` setters in display server ([GH-102719](https://github.com/godotengine/godot/pull/102719)).
```

> - Physics: [Jolt Physics] Fix ghost collision issue on dense triangle meshes ([GH-102614](https://github.com/godotengine/godot/pull/102614)).
> - Physics: [Jolt Physics] Improve performance of certain physics queries ([GH-101071](https://github.com/godotengine/godot/pull/101071)).

Reviewer 2: ```suggestion
- Physics: Jolt: Fix ghost collision issue on dense triangle meshes ([GH-102614](https://github.com/godotengine/godot/pull/102614)).
- Physics: Jolt: Improve performance of certain physics queries ([GH-101071](https://github.com/godotengine/godot/pull/101071)).
```

- Porting: Add support for embedding game process in the Android Editor ([GH-102492](https://github.com/godotengine/godot/pull/102492)).

> - Porting: Fix Game and editor freeze when clicking on the game's title bar ([GH-102744](https://github.com/godotengine/godot/pull/102744)).

Reviewer 2: ```suggestion
- Porting: Fix game and editor freeze when clicking on the game's title bar ([GH-102744](https://github.com/godotengine/godot/pull/102744)).
```
